### PR TITLE
Api 58 sovereign token prefix

### PR DESCRIPTION
**Pull request by bogdan-rosianu**
## Reasoning
- 
- 
- 
  
## Proposed Changes
- 
- 
- 

## How to test
- 
- 
-

### CODE REVIEW

File: src/endpoints/esdt/esdt.address.service.ts [hunk 1]
```diff
@@ -217,14 +216,14 @@ export class EsdtAddressService {
...
       const identifier = filter.identifiers[0];
 
-      if (identifier.split('-').length === 2) {
+      if (TokenUtils.isSovereignIdentifier(identifier)) {
+        collection = identifier.split('-').slice(0, 3).join('-');
+        nonceHex = identifier.split('-')[3];
+      } else {
         collection = identifier.split('-').slice(0, 2).join('-');
         nonceHex = identifier.split('-')[2];
       }
```
Comment (cfaur09) [inline src/endpoints/esdt/esdt.address.service.ts:225]: you can avoid identifier.split('-') by adding into a const like:

const parts = identifier.split('-')

File: src/endpoints/esdt/esdt.address.service.ts [hunk 2]
```diff
@@ -268,7 +268,11 @@ export class EsdtAddressService {
     for (const dataSourceNft of nfts) {
       const nft = new NftAccount();
       nft.identifier = dataSourceNft.tokenIdentifier;
-      nft.collection = dataSourceNft.tokenIdentifier.split('-').slice(0, 2).join('-');
+      if (TokenUtils.isSovereignIdentifier(dataSourceNft.tokenIdentifier)) {
```
Comment (cfaur09) [inline src/endpoints/esdt/esdt.address.service.ts:271]: same as above

File: src/endpoints/esdt/esdt.address.service.ts (hunk 1 above)
Comment (bogdan-rosianu) [reply src/endpoints/esdt/esdt.address.service.ts]: fixed

File: src/endpoints/esdt/esdt.address.service.ts (hunk 2 above)
Comment (bogdan-rosianu) [reply src/endpoints/esdt/esdt.address.service.ts:273]: fixed

**Review by cfaur09 — APPROVED** [review]

**Pull request merged**